great_expectations/checkpoint/checkpoint.py

```diff
@@ -117,6 +118,19 @@ class Config:
             Renderer: lambda r: r.serialize(),
         }
 
+    @override
+    def json(self, **kwargs: Any) -> str:
+        # Necessary override to check that all children validation definitions are added.
+        # Without this, JSON encoder will raise an error on the first non-added child.
+        diagnostics = self.is_added()
+        if not diagnostics.is_added:
+            if not diagnostics.parent_added and diagnostics.children_added:
+                pass  # Checkpoint is not added but all children are - valid serialization state
+            else:
+                diagnostics.raise_for_error()
+
+        return super().json(**kwargs)
+
```

Comment:
@Kilo59 is there a better way to do this? The JSON encoder will fail on the first error here I want to aggregate all the errors.